### ISSUE TITLE
Enhancement:  sort by domain name in addition to site name

### ISSUE THREAD

**Issue by banyudu** [DeliciousGarfield/tabmanager#1]
Some websites has sub-websites, for example _en.wikipedia.org_ and _zh.wikipedia.org_ are both sub-websites of wikipedia.org.

When sort by site names, _en.wikipedia.org_ and _zh.wikipedia.org_ cannot sort together. 

I suggest sort with domain name (_wikipedia.org_) firstlly, then site name (_en.wikipedia.org_, _zh.wikipedia.org_) if domain names are equal.

**Comment by DeliciousGarfield** [comment]
@banyudu Well, that's a good idea, I have considered sorting by subdomains too. But there is a little problem in imlementation process. It's a little trivial to distinguish subdomains, as not all websites have the tipical three-part domain names. 
For example, there are many irregular domain names, like `news.sina.com.cn` which is a four-part domain name, and `stackoverflow.com` which is a two-part domain name. I have to distinguish which is the main domain, and which is subdomain. So I decide to solve it later.
A possible solution seems to be split the domain name to two parts, separated by the first dot. Well, I think that's enough. Forget what I said above:) But that's not enough for domains like `match.sports.sina.com.cn` and `sports.sina.com.cn`

**Comment by banyudu** [comment]
@DeliciousGarfield  

You need not to split site name to parts by dot. What we care is the suffix (_.com.cn_, _.com_, _.org_) and the last non-suffix word. So you can save a list of known suffixes to extract the exact domain name from site name.

For example, "news.sina.com.cn" has known suffixes ".com.cn" and ".cn", we use the longest one as real suffix, and the last non-suffix word is sina, so its domain name is "sina" + ".com.cn" -> "sina.com.cn". 

As for "stackoverflow.com", its suffix is ".com", and last non-suffix word is "stackoverflow", so the domain name is "stackoverflow.com".

**Comment by DeliciousGarfield** [comment]
You are right. The `list of known domain name suffixes` is what bothers me. At first I want to cover all suffixes, but now I think that is difficult and unnecessary, The most common `list of known domain name suffixes` will be enough.
